spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -1141,10 +1142,22 @@ struct ProductionVisitor
         };
 
         // Handle synchronization via `synchronize-at` or `synchronize-after` unit properties.
-        if ( const auto& unit = p->tryAs<production::Unit>() ) {
-            const auto& type = unit->unitType();
-            const auto synchronize_at = type.propertyItem("%synchronize-at");
-            const auto synchronize_after = type.propertyItem("%synchronize-after");
+        // We can either see a unit for synchronization in a list (generating a
+        // `while` production), or directly.
+        std::optional<type::Unit> unitType;
+        if ( while_ ) {
+            if ( const auto& field = while_->meta().field() )
+                if ( auto unit = field->parseType().elementType().tryAs<type::Unit>() )
+                    unitType = *unit;
+        }
+
+        else if ( const auto& unit = p->tryAs<production::Unit>() ) {
+            unitType = unit->unitType();
+        }
```

Comment:
```suggestion
        else if ( const auto& unit = p->tryAs<production::Unit>() )
            unitType = unit->unitType();
```

Reply:
I actually thought of that, but incorrectly went with symmetry with the first case ;). Fixed.